### DISCUSSION TITLE
/showing-or-hiding-sharepoint-framework-listview-custom-actions-based-on-permissions-and-selected-items/

### DISCUSSION THREAD

**Discussion by comment-migrator** [estruyf/eliostruyf-comments#271] [Comments]
Showing or hiding SharePoint Framework ListView custom actions based on permissions and selected items
        
[https://www.eliostruyf.com/showing-or-hiding-sharepoint-framework-listview-custom-actions-based-on-permissions-and-selected-items/](https://www.eliostruyf.com/showing-or-hiding-sharepoint-framework-listview-custom-actions-based-on-permissions-and-selected-items/)

**Comment by comment-migrator** [comment]
```header
Originally posted by resing on 2017-07-07 23:09
```

Excellent examples. Love the visuals. I didn't know about this before and after a quick scan I feel I know something new! Looks very similar to the way the Flow and PowerApps teams are using these extensions.

**Reply by comment-migrator** [reply]
```header
Originally posted on 2017-07-08 17:37
```

Thanks!

**Comment by comment-migrator** [comment]
```header
Originally posted by Paul Matthews on 2017-12-04 17:28
```

Has this changed since you have written this post? I'm unable to find IListViewCommandSetRefreshEventParameters. Also is it possible to perform a REST Call to get a result before displaying the visibility of the button? When I have tried, I'm unable to change the visibility of the button after the callback.

**Reply by comment-migrator** [reply]
```header
Originally posted on 2018-01-31 10:28
```

Sorry, I missed this, the APIs have indeed changed a bit. **onRefreshCommand** does not exist anymore, this is now changed to **onListViewUpdated**. I have updated the code snippets and will updated the article as well. Thanks for mentioning this!

**Comment by comment-migrator** [comment]
```header
Originally posted by Martin Coupal on 2018-04-26 00:12
```

Hello Elio, do you know if it is possible to change the action on an existing command (ex: change the behavior of the New command) with Framework Extension? There is a major issue with new and edit forms displayed as sliding panel as you will loose all your updates if you click outside of the panel without any warning. I would like to change the new command behavior to redirect to the NewForm.aspx page. This will launch as a full page without the issue of the sliding panel. Note: I tried to customized the form using PowerApps but have the same issue as it launches in the same sliding panel. Thanks for all your great articles!

**Reply by comment-migrator** [reply]
```header
Originally posted on 2018-04-26 09:02
```

Thanks for the feedback, but currently there is no supported way to override the existing ones.

This might be a good thing to suggest in the uservoice: [https://sharepoint.uservoic...](https://sharepoint.uservoice.com/forums/329220-sharepoint-dev-platform "https://sharepoint.uservoice.com/forums/329220-sharepoint-dev-platform")

**Comment by comment-migrator** [comment]
```header
Originally posted by Marco Pagani on 2019-10-24 17:30
```

Hello Elio, I realized that the onListViewUpdated method is not called if I have already selected an element and I select another one by clicking outside the checkbox, on the columns.  
did you have this problem?  
Thanks!

**Comment by comment-migrator** [comment]
```header
Originally posted by MaxE87 on 2020-06-22 22:42
```

Hi @eliostruyf:disqus thank you for the nice article, I really like it. Do you know, if there is an easy way to show the button not based on list permissions but based on the permission of the items in the list?

**Comment by comment-migrator** [comment]
```header
Originally posted by max1415892 on 2020-06-23 13:53
```

Hi @eliostruyf:disqus thank you for the nice article! Do you know if it is possible to show the button only if the user has the permisson to edit a selected item? My first approach was to make an api call but it seems, the onListViewUpdated method cannot be rewritten to an async function.

**Comment by comment-migrator** [comment]
```header
Originally posted by Samuel Molahloe on 2020-09-28 13:57
```

Elio, how does one go about disabling/enabling the CommandSet instead of hiding it.

**Comment by EA12** [comment]
Hi Elio,

nice writing, as always.
You may want to add the import declaration for SPPermission. Beginners might struggling with searching for it...

import { SPPermission } from '@microsoft/sp-page-context';  

Thanks and all the Best, Ronny
